code/controller/actuator.cpp

```diff
@@ -152,15 +157,17 @@ void Actuator::moveActuator(const unsigned char device, const int value, const i
         for (int i = value; i > 0; i--) {
             if (m_serial_port->isOpen()) {
                 m_serial_port->write(instr, CMD_SIZE + DATA_SIZE);
-            }
-            else {
+            } else {
                 Logger::log("ERROR: Failed to write to serial port " + (m_serial_port->portName()).toStdString() + " because it's not open.", Logger::ERROR);
                 throw "Action could not be completed"; // TODO: Might want to figure out a better way than throwing exceptions, revisit after adding concurrency
             }
 
             std::this_thread::sleep_for(sleep_step);
         }
     }
+    catch (char const *e) {
+        throw e;
+    }
     catch (...) {

```

Comment:
Tried a variety of things but just throwing std::current_exception() and then catching std::exception& was, while not outright crashing the program, still trigger and error and force it to close.

I've left in the multiple catch until we can figure something else out